remote-api/src/main/java/org/alfresco/rest/api/impl/CategoriesImpl.java

```diff
@@ -106,6 +102,37 @@ public List<Category> createSubcategories(String parentCategoryId, List<Category
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public CollectionWithPagingInfo<Category> getCategoryChildren(String parentCategoryId, Parameters params)
+    {
+        final NodeRef parentNodeRef = getParentNodeRef(parentCategoryId);
+        final List<ChildAssociationRef> childCategoriesAssocs = nodeService.getChildAssocs(parentNodeRef).stream()
+                        .filter(ca -> ContentModel.ASSOC_SUBCATEGORIES.equals(ca.getTypeQName()))
+                        .collect(Collectors.toList());
+        final List<Category> categories = childCategoriesAssocs.stream()
+                .map(c -> mapToCategory(c.getChildRef()))
+                .collect(Collectors.toList());
+        return ListPage.of(categories, params.getPaging());
+    }
+
+    private NodeRef getCategoryNodeRef(String nodeId)
+    {
+        final NodeRef nodeRef = nodes.validateNode(nodeId);
+        if (isNotACategory(nodeRef))
+        {
+            throw new InvalidArgumentException(NOT_A_VALID_CATEGORY, new String[]{nodeId});
+        }
+        return nodeRef;
+    }
+
+    private NodeRef getParentNodeRef(String parentCategoryId)
```

Comment:
I assume you didn't call this `getCategoryNode` because the root isn't a category? Perhaps it would be clearer being called `getCategoryHierarchyNode`? Or alternatively if you just added some javadoc to explain the purpose (in particular I'd expect `getParentNodeRef(id)` to give me the parent of the supplied node.

Reply:
I had trouble to name this method so that it is not confusing.
The general rule here is:
- `getParentNodeRef` is called when we start off and pass parent category id in the call (if that id is not `-root` then it simply returns `getCategoryNodeRef`)
- `getCategoryNodeRef` is called when we start off and pass category id in the call (for now it is for get single category)
Maybe I could merge these methods into one, WDYT?

Reply:
Actually that `getParentNodeRef(id)` was meant to handle a case when `-root-` is explicitly passed as a parent category id.
I am pushing refactored code along with some javadocs - hopefully this will clarify this.